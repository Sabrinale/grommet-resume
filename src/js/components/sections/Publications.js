import React from 'react'

import { formatDate } from '../common'
import { SecondaryTitle } from '../SecondaryTitle'
import { Description } from '../Description'
import { Highlights } from '../Highlights'
import { SectionTitle } from '../SectionTitle'
import { DateSection } from '../DateSection'

const Publication = ({ item }) => (
  <Highlights>
    <SectionTitle>{item.name}</SectionTitle>
    <SecondaryTitle>{item.publisher}</SecondaryTitle>
    <DateSection>{formatDate(item.releaseDate)}</DateSection>
    <Description>{item.summary}</Description>
  </Highlights>
)

export { Publication }
