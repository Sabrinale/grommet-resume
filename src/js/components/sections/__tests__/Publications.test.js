import React from 'react'
import renderer from 'react-test-renderer'

import { Publication } from '../Publications'

import defaultResume from './mockResume.json'

test('renders publications section', () => {
  const data = defaultResume.publications[0]
  const component = renderer.create(<Publication item={data} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
