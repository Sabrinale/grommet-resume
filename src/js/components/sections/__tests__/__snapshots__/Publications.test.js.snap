// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders publications section 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 kFdcfG"
>
  <h3
    className="StyledHeading-sc-1rdh4aw-0 jVEgcZ"
  >
    Publication
  </h3>
  <h4
    className="StyledHeading-sc-1rdh4aw-0 hlZfma"
  >
    Company
  </h4>
  <span
    className="StyledText-sc-1sadyjn-0 dndifk"
    size="medium"
  >
    Oct 2014
  </span>
  <p
    className="StyledParagraph-tbetod-0 jzEULC"
  >
    Description...
  </p>
</div>
`;
